src/components/sectionList/bulk/BulkSharing.tsx

```diff
@@ -0,0 +1,269 @@
+import { useAlert } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { Button, Divider, Field, NoticeBox } from '@dhis2/ui'
+import React, { FormEvent, useState } from 'react'
+import { useSchemaFromHandle } from '../../../lib'
+import {
+    SharingSearchSelect,
+    MetadataAccessField,
+    SharingSearchResult,
+    DataAccessField,
+} from '../../sharing'
+import css from './Bulk.module.css'
+import { ActionSummary } from './BulkActionSummary'
+import {
+    SharingJsonPatchOperation,
+    useBulkSharingMutation,
+} from './useBulkSharing'
+
+type BulkSharingProps = {
+    selectedModels: Set<string>
+    onSaved: () => void
+    children: ({
+        submitting,
+        disableSave,
+    }: {
+        submitting: boolean
+        disableSave: boolean
+    }) => React.ReactNode
+}
+
+export type SharingAction = {
+    op: 'remove' | 'replace'
+    sharingEntity: SharingSearchResult
+    access: string
+}
+
+const actionToJsonPatchOperation = (
+    action: SharingAction
+): SharingJsonPatchOperation => {
+    const { op, sharingEntity, access } = action
+    const value = {
+        access: access,
+        id: sharingEntity.id,
+    }
+
+    return {
+        op,
+        path: `/sharing/${sharingEntity.entity}/${sharingEntity.id}`,
+        ...(op === 'remove' ? undefined : { value }),
+    }
+}
+
+/*
+   Usage of React-Final-Form does not seem necessary because we're not using 
+   validation or initialState. And the result of the form (list of added SharingActions) are not kept in form-state. 
```

Comment:
I started implementing RFF, however, to implement this in the form-state we would've needed another dependency on https://github.com/final-form/react-final-form-arrays?tab=readme-ov-file#simple-example , so I decided to just keep it simple. However it means I've implemented some generic form-state here.